src/storage/rpc.rs

```diff
@@ -103,13 +105,24 @@ impl<N: Network> Storage for RpcStorage<N> {
                 return Ok(None);
             }
             let code = Bytecode::new_raw(code);
+            let code_hash = if code.is_empty() {
+                None
+            } else {
+                Some(code.hash_slow())
+            };
+            if let Some(code_hash) = code_hash {
+                self.cache_bytecodes
+                    .lock()
+                    .unwrap()
+                    .insert(code_hash, code.into());
+            }
             let basic = AccountBasic { balance, nonce };
             self.cache_accounts.lock().unwrap().insert(
                 *address,
                 EvmAccount {
                     basic: basic.clone(),
-                    code_hash: (!code.is_empty()).then(|| code.hash_slow()),
-                    code: (!code.is_empty()).then(|| code.into()),
+                    code_hash,
+                    code: None,
```

Comment:
We now don't snapshot new codes?

Reply:
good catch, we should add and use `pub fn get_cache_bytecodes`

Reply:
resolved in https://github.com/risechain/pevm/pull/248